src/khoj/processor/conversation/gpt4all/utils.py

```diff
@@ -16,8 +17,13 @@ def download_model(model_name: str):
 
     # Decide whether to load model to GPU or CPU
     try:
-        # Check if machine has GPU and GPU has enough free memory to load the chat model
-        device = "gpu" if gpt4all.pyllmodel.LLModel().list_gpu(chat_model_config["path"]) else "cpu"
+        # Try load chat model to GPU if:
+        # 1. Loading chat model to GPU isn't disabled via CLI and
+        # 2. Machine has GPU
+        # 3. GPU has enough free memory to load the chat model
+        device = (
+            "gpu" if state.chat_on_gpu and gpt4all.pyllmodel.LLModel().list_gpu(chat_model_config["path"]) else "cpu"
```

Comment:
What does `gpt4all.pyllmodel.LLModel().list_gpu(chat_model_config["path"])` do? Would that list available GPUs?

Reply:
Yeah this code just checks if any GPU(s) on the host machine has enough memory etc. to load the chat model before attempting to load it onto GPU. 

Note: This change is not part of this PR's, it's just a merge with master